processor/src/test/java/org/mapstruct/ap/test/nestedtargetproperties/NestedTargetPropertiesTest.java

```diff
@@ -125,4 +143,67 @@ public void shouldReverseNestedTarget() {
         assertThat( result.getRecordedAt() ).isEqualTo( "Live, First Avenue, Minneapolis" );
         assertThat( result.getSongTitle() ).isEqualTo( "Purple Rain" );
     }
+
+    @Test
+    @WithClasses({ChartEntryToArtistUpdate.class})
+    public void shouldMapNestedTargetWitUpdate() {
+
+        ChartEntry chartEntry = new ChartEntry();
+        chartEntry.setArtistName( "Prince" );
+        chartEntry.setChartName( "US Billboard Hot Rock Songs" );
+        chartEntry.setCity( "Minneapolis" );
+        chartEntry.setPosition( 1 );
+        chartEntry.setRecordedAt( "Live, First Avenue, Minneapolis" );
+        chartEntry.setSongTitle( null );
+
+        Chart result = new Chart();
+        result.setSong( new Song() );
+        result.getSong().setTitle( "Raspberry Beret" );
+
+        ChartEntryToArtistUpdate.MAPPER.map( chartEntry, result );
+
+        assertThat( result.getName() ).isEqualTo( "US Billboard Hot Rock Songs" );
+        assertThat( result.getSong() ).isNotNull();
+        assertThat( result.getSong().getArtist() ).isNotNull();
+        assertThat( result.getSong().getTitle() ).isEqualTo( "Raspberry Beret" );
+        assertThat( result.getSong().getArtist().getName() ).isEqualTo( "Prince" );
+        assertThat( result.getSong().getArtist().getLabel() ).isNotNull();
+        assertThat( result.getSong().getArtist().getLabel().getStudio() ).isNotNull();
+        assertThat( result.getSong().getArtist().getLabel().getStudio().getName() )
+            .isEqualTo( "Live, First Avenue, Minneapolis" );
+        assertThat( result.getSong().getArtist().getLabel().getStudio().getCity() )
+            .isEqualTo( "Minneapolis" );
+        assertThat( result.getSong().getPositions() ).hasSize( 1 );
+        assertThat( result.getSong().getPositions().get( 0 ) ).isEqualTo( 1 );
+
+    }
+
+    @Test
+    @WithClasses({FishTankMapper.class})
```

Comment:
You know my CPU isn't that powerful, so moving all mappers to the top of the class would really help... For debugging purposes, I usually remove those mappers that I'm not interested in from the `WithClasses`... 😄